libimage/runtime.go

```diff
@@ -204,6 +211,25 @@ func (r *Runtime) LookupImage(name string, options *LookupImageOptions) (*Image,
 		name = strings.TrimPrefix(name, "sha256:")
 	}
 
+	// Set the platform for matching local images.
+	if !options.IgnorePlatform {
+		if options.Architecture == "" {
+			options.Architecture = r.systemContext.ArchitectureChoice
+		}
+		if options.Architecture == "" {
+			options.Architecture = runtime.GOARCH
```

Comment:
This worries me a bit because it loses the distinction between runtime environment and user request: in particular if this gets to c/image (as it seems to via `manifestList.LookupInstance`, it turns off variant auto-detection and blindly accepts all variants AFAICS.